src/opm/output/eclipse/AggregateMSWData.cpp

```diff
@@ -610,6 +626,72 @@ namespace {
                 usys.from_si(M::length, sicd.length());
         }
 
+        template <class RSegArray>
+        void assignAICDCharacteristics(const ::Opm::Segment&    segment,
+                                            const ::Opm::UnitSystem& usys,
+                                            const int                baseIndex,
+                                            RSegArray&               rSeg)
+        {
+            using Ix = ::Opm::RestartIO::Helpers::VectorItems::RSeg::index;
+            using M  = ::Opm::UnitSystem::measure;
+
+            const auto& aicd = segment.autoICD();
+
+            rSeg[baseIndex + Ix::DeviceBaseStrength] =
+                usys.from_si(M::aicd_strength, aicd.strength());
+
+            rSeg[baseIndex + Ix::ScalingFactor] = ((aicd.methodFlowScaling() == 1) ||
+             ((aicd.methodFlowScaling() < 0) && (aicd.length() < 0))) ?
+                usys.from_si(M::length, aicd.scalingFactor()) : aicd.scalingFactor();
+
+            rSeg[baseIndex + Ix::CalibrFluidDensity] =
+                usys.from_si(M::density, aicd.densityCalibration());
+
+            rSeg[baseIndex + Ix::CalibrFluidViscosity] =
+                usys.from_si(M::viscosity, aicd.viscosityCalibration());
+
+            rSeg[baseIndex + Ix::CriticalWaterFraction] = aicd.criticalValue();
+
+            rSeg[baseIndex + Ix::TransitionRegWidth] =
+                aicd.widthTransitionRegion();
+
+            rSeg[baseIndex + Ix::MaxEmulsionRatio] =
+                aicd.maxViscosityRatio();
+
+            rSeg[baseIndex + Ix::FlowRateExponent] =
+                aicd.flowRateExponent();
+
+            rSeg[baseIndex + Ix::ViscFuncExponent] =
+                aicd.viscExponent();
+
+            rSeg[baseIndex + Ix::MaxValidFlowRate] =
+            (aicd.maxAbsoluteRate() == std::numeric_limits<double>::max()) ?
+               -2.e+20 : usys.from_si(M::geometric_volume_rate, aicd.maxAbsoluteRate()) ;
```

Comment:
Magic number alert!

Reply:
Se previous comment, Eclipse often uses these special numeric values for various default cases, this is one of the values used.